test/json_test/CMakeLists.txt

```diff
@@ -1,43 +1,69 @@
 include(CTest)
 
-set(INC_DIR "${PROJECT_SOURCE_DIR}/include")
 set(GEN_DIR "${CMAKE_CURRENT_BINARY_DIR}/generated")
 set(FBS_DIR "${PROJECT_SOURCE_DIR}/test/monster_test")
 
 set(DATA_DST "${CMAKE_CURRENT_BINARY_DIR}")
 set(DATA_SRC "${PROJECT_SOURCE_DIR}/test/flatc_compat")
 
-include_directories("${GEN_DIR}" "${INC_DIR}")
+include_directories("${GEN_DIR}")
 
-add_custom_target(gen_monster_test_json ALL)
+# FIXME: possible improvement if cmake_minimum_required >= 3.1:
+#        use add_library(... INTERFACE) +  add_custom_command + target_sources(INTERFACE) + target_link_libraries()
+
+# Use add_custom_target + add_custom_command + add_dependencies because a parallel MSVC build will cause
+# add_custom_command to be run multiple times in parallel which causes errors opening file for write.
+# This problem does not pose when generated sources are used by only one target.
+add_custom_target (gen_json_test
```

Comment:
@mikkelfj 
I suppose you don't want to have these sources hardcoded on master.
So I propose, if/when you accept these changes, to merge these into some `cmake_work` branch on which @wdobbe can base his pull request.
And only if everything falls together, you can merge into master.

Reply:
I could make a branch, but I suppose @wdobbe might just as well use your branch directly from the PR?
I will probably squash it all before it lands.